src/GovUk.Education.ExploreEducationStatistics.Admin/Services/DataSetScreenerClient.cs

```diff
@@ -0,0 +1,37 @@
+using GovUk.Education.ExploreEducationStatistics.Admin.Requests;
+using GovUk.Education.ExploreEducationStatistics.Admin.Services.Interfaces;
+using GovUk.Education.ExploreEducationStatistics.Common.Model;
+using System;
+using System.Net.Http;
+using System.Net.Http.Json;
+using System.Text;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace GovUk.Education.ExploreEducationStatistics.Admin.Services;
+
+public class DataSetScreenerClient(IHttpClientFactory httpClientFactory) : IDataSetScreenerClient
+{
+    private readonly JsonSerializerOptions _jsonSerializerOptions = new()
+    {
+        PropertyNamingPolicy = JsonNamingPolicy.CamelCase, // API is case sensitive
+    };
+
+    public async Task<DataSetScreenerResponse> ScreenDataSet(
+        DataSetScreenerRequest dataSetRequest,
+        CancellationToken cancellationToken)
+    {
+        var json = JsonSerializer.Serialize(dataSetRequest, _jsonSerializerOptions);
+        var content = new StringContent(json, Encoding.UTF8, "application/json");
+
+        var client = httpClientFactory.CreateClient("DataSetScreener");
+        // TODO (EES-5353): Add cancellation token handling logic to terminate Azure Function processes
+        // TODO (EES-5999): Replace hardcoded URL with appsetting
+        var response = await client.PostAsync("http://localhost/api/screen", content, CancellationToken.None);
```

Comment:
I think IHttpClient has a `PostAsJsonAsync()` method on it that would do the JSON serialisation and `application/json` request headers for you.  Not sure if it'll let you supply the camelCase behaviour through settings or not though?

I think as well that you can register the `HttpClientFactory` in Startup and just request an `HttpClient` be injected into `DataSetScreenerClient`.  See `ProcessorClient` for an example of all these things.